cpp/daal/src/externals/service_memory.h

```diff
@@ -138,6 +139,18 @@ void service_memset_seq(T * const ptr, const T value, const size_t num)
     }
 }
 
+/* Initialize block of memory of length num with entries [startValue, ..., startValue + num -1]*/
+template <typename T, CpuType cpu>
+void service_memset_incrementing(T * const ptr, const T startValue, const size_t num)
+{
+    PRAGMA_IVDEP
+    PRAGMA_VECTOR_ALWAYS
+    for (size_t i = 0; i < num; i++)
```

Comment:
```suggestion
    for (size_t i = 0; i < num; ++i)
```

It should not affect performance because of how smart compilers are. But it still will help to maintain consistency.

Reply:
Btw we can consider using `std::iota`. I believe using standard library bits is generally not approved but in this case it is somewhat justified because it is pure header only functionality. We can think about it later as well as part of our strategy for simplification of the DAAL code maintenance.

Reply:
`i++` is used everywhere in `service_memory.h`. Should I replace all occurences?